statefun-e2e-tests/statefun-exactly-once-e2e/src/test/resources/wrapped-messages-ingress-module/module.yaml

```diff
@@ -0,0 +1,36 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: "1.0"
+
+module:
+  meta:
+    type: remote
+  spec:
+    ingresses:
```

Comment:
One main reason for not using YAML-based egress is because it requires an extra dependency on `statefun-flink-io` for the `KafkaProducerRecord` class - the Java function needs to be sending records of type `KafkaProducerRecord` to the egress, as that's the type the YAML generic Kafka egress expects.

In short, as of now, it's a bit awkward trying to send messages to YAML-based Kafka egress from Java functions.

Reply:
On the other hand, there is no major reason to not use Java for the ingress.

The only reason I used YAML-based Kafka ingress was because it was easier, without the need to write any serializer or router :)

But for consistency, I will move the ingress to also be in the Java embedded module.